scripts/validate_min_versions_in_sync.py

```diff
@@ -41,6 +50,130 @@
 import _optional
 
 
+def pin_min_versions_to_ci_deps():
+    """
+    Pin minimum versions to CI dependencies.
+
+    Pip dependencies are not pinned.
+    """
+    all_yaml_files = list(YAML_PATH.iterdir())
+    all_yaml_files.append(ENV_PATH)
+    toml_dependencies = {}
+    with open(SETUP_PATH, "rb") as toml_f:
+        toml_dependencies = tomllib.load(toml_f)
+    for curr_file in all_yaml_files:
+        with open(curr_file) as yaml_f:
+            yaml_file_data = yaml_f.read()
+            yaml_file = yaml.safe_load(yaml_file_data)
+            yaml_dependencies = yaml_file["dependencies"]
+            yaml_map = get_yaml_map_from(yaml_dependencies)
+            toml_map = get_toml_map_from(toml_dependencies)
+            with open(curr_file, "w") as f:
+                f.write(
+                    pin_min_versions_to_yaml_file(yaml_map, toml_map, yaml_file_data)
```

Comment:
Since this runs on every commit, we don't necessarily want to rewrite this file every time.